File: scribbleshare-app/scribbleshare-frontend/src/scripts/protocol/server/messages/ServerMessageAddUser.ts
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";
import ByteBuffer from "../../ByteBuffer.js";

export default class ServerMessageAddUser extends ServerMessage {
    user: bigint;

    constructor(byteBuffer: ByteBuffer) {
        super();
        this.user = byteBuffer.readBigInt64();
    }

    getType(): ServerMessageType {
        return ServerMessageType.ADD_USER;
    }
}

File: room/src/main/java/net/stzups/scribbleshare/room/server/websocket/protocol/server/messages/ServerMessageAddUser.java
package net.stzups.scribbleshare.room.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.scribbleshare.data.objects.User;
import net.stzups.scribbleshare.room.server.websocket.protocol.server.ServerMessage;
import net.stzups.scribbleshare.room.server.websocket.protocol.server.ServerMessageType;
import net.stzups.util.DebugString;

public class ServerMessageAddUser extends ServerMessage {
    private final User user;

    public ServerMessageAddUser(User user) {
        this.user = user;
    }

    @Override
    protected ServerMessageType getMessageType() {
        return ServerMessageType.ADD_USER;
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        byteBuf.writeLong(user.getId());
    }

    @Override
    public String toString() {
        return DebugString.get(ServerMessageAddUser.class)
                .add("user", user)
                .toString();
    }
}
